ush/python/pygfs/task/land_analysis.py

```diff
@@ -87,12 +87,13 @@ def prepare_GTS(self) -> None:
         logger.info("Copying GTS obs for bufr2ioda.x")
         FileHandler(prep_gts_config.gtsbufr).sync()
 
-        # generate bufr2ioda YAML file
-        bufr2ioda_yaml = os.path.join(self.runtime_config.DATA, "bufr_adpsfc_snow.yaml")
-        logger.info(f"Generate BUFR2IODA YAML file: {bufr2ioda_yaml}")
-        temp_yaml = parse_j2yaml(self.task_config.BUFR2IODAYAML, self.task_config)
-        save_as_yaml(temp_yaml, bufr2ioda_yaml)
-        logger.info(f"Wrote bufr2ioda YAML to: {bufr2ioda_yaml}")
+        # generate bufr2ioda YAML files
+        for name in ["adpsfc", "snocvr"]:
+            gts_yaml = os.path.join(self.runtime_config.DATA, f"bufr_{name}_snow.yaml")
+            logger.info(f"Generate BUFR2IODA YAML file: {gts_yaml}")
+            temp_yaml = parse_j2yaml(prep_gts_config.bufr2ioda[name], localconf)
+            save_as_yaml(temp_yaml, gts_yaml)
+            logger.info(f"Wrote bufr2ioda YAML to: {gts_yaml}")
```

Comment:
Thanks for updating this section @jiaruidong2017 
Here is what I was suggesting.  
In the attached patch, you will notice that I have removed any knowledge of what is being converted from  bufr to ioda in the python script.  The control comes directly from the yaml.
[diff.txt](https://github.com/NOAA-EMC/global-workflow/files/12580213/diff.txt)

Would you be willing to give this a try?
Thanks

Reply:
Thanks @aerorahul. I will try it and let you know then.

Reply:
@aerorahul I just tested and it worked perfectly. The changes have been committed to the repo.